pkg/controller/telemetry/metric.go

```diff
@@ -502,156 +502,158 @@ func (m *MetricController) Run(ctx context.Context, mapOfTcpInfo *ebpf.Map) {
-	data.srtt = connectData.statistics.SRttTime
-	data.minRtt = connectData.statistics.RttMin
-	data.totalRetrans = connectData.statistics.Retransmits - tcpConns[data.conSrcDstInfo].totalRetrans
-	data.packetLost = connectData.statistics.LostPackets - tcpConns[data.conSrcDstInfo].packetLost
-
-	cm, ok := tcpConns[data.conSrcDstInfo]
+	reqMetric.sentBytes = rawStats.SentBytes - tcpConns[reqMetric.conSrcDstInfo].sentBytes
+	reqMetric.receivedBytes = rawStats.ReceivedBytes - tcpConns[reqMetric.conSrcDstInfo].receivedBytes
+	reqMetric.state = rawStats.State
+	reqMetric.success = rawStats.ConnectSuccess
+	reqMetric.duration = rawStats.Duration
+	reqMetric.startTime = rawStats.StartTime
+	reqMetric.lastReportTime = rawStats.LastReportTime
+	reqMetric.srtt = rawStats.statistics.SRttTime
+	reqMetric.minRtt = rawStats.statistics.RttMin
+	reqMetric.totalRetrans = rawStats.statistics.Retransmits - tcpConns[reqMetric.conSrcDstInfo].totalRetrans
+	reqMetric.packetLost = rawStats.statistics.LostPackets - tcpConns[reqMetric.conSrcDstInfo].packetLost
+
+	cm, ok := tcpConns[reqMetric.conSrcDstInfo]
 	if ok {
-		cm.receivedBytes = connectData.ReceivedBytes
-		cm.sentBytes = connectData.SentBytes
-		cm.totalRetrans = connectData.statistics.Retransmits
-		cm.packetLost = connectData.statistics.LostPackets
+		cm.receivedBytes = rawStats.ReceivedBytes
+		cm.sentBytes = rawStats.SentBytes
+		cm.totalRetrans = rawStats.statistics.Retransmits
+		cm.packetLost = rawStats.statistics.LostPackets
 		cm.totalReports++
-		tcpConns[data.conSrcDstInfo] = cm
+		tcpConns[reqMetric.conSrcDstInfo] = cm
 	} else {
-		tcpConns[data.conSrcDstInfo] = connMetric{
-			receivedBytes: connectData.ReceivedBytes,
-			sentBytes:     connectData.SentBytes,
-			totalRetrans:  connectData.statistics.Retransmits,
-			packetLost:    connectData.statistics.LostPackets,
+		tcpConns[reqMetric.conSrcDstInfo] = connMetric{
+			receivedBytes: rawStats.ReceivedBytes,
+			sentBytes:     rawStats.SentBytes,
+			totalRetrans:  rawStats.statistics.Retransmits,
+			packetLost:    rawStats.statistics.LostPackets,
 			totalReports:  1,
 		}
 	}
 
-	return data, nil
+	return reqMetric, nil
 }
 
 func buildV6Metric(buf *bytes.Buffer, tcpConns map[connectionSrcDst]connMetric) (requestMetric, error) {
-	data := requestMetric{}
-	connectData := connectionDataV6{}
-	if err := binary.Read(buf, binary.LittleEndian, &connectData); err != nil {
-		return data, err
+	reqMetric := requestMetric{}
+	rawStats := connectionDataV6{}
+	if err := binary.Read(buf, binary.LittleEndian, &rawStats); err != nil {
+		return reqMetric, err
 	}
-	data.conSrcDstInfo.src = connectData.SrcAddr
-	data.conSrcDstInfo.dst = connectData.DstAddr
-	data.direction = connectData.Direction
-	data.conSrcDstInfo.dstPort = connectData.DstPort
-	data.conSrcDstInfo.srcPort = connectData.SrcPort
+	reqMetric.conSrcDstInfo.src = rawStats.SrcAddr
+	reqMetric.conSrcDstInfo.dst = rawStats.DstAddr
+	reqMetric.conSrcDstInfo.direction = rawStats.Direction
```

Comment:
Is this the root cause of the problem? Too many renames swamping critical changes.

Reply:
right